apps/portal-api/src/modules/services/document/document-download-endpoint.ts

```diff
@@ -44,25 +57,51 @@ export const documentDownloadEndpoint = (app) => {
       documentDownloadRateLimiter,
       async (req: Request, res) => {
         const { attach } = req.query;
-        const user = await loadUser(req);
+        const { user, isLoadedFromUserPlatformToken } = await loadUser(req);
         if (!user) {
           res.status(401).json({ message: 'You must be logged in' });
           return;
         }
 
+        const context: PortalContext = {
+          user: user,
+          serviceInstanceId: extractId<ServiceInstanceId>(
+            req.params.serviceInstanceId
+          ),
+          req,
+          res,
+        };
+
+        if (isLoadedFromUserPlatformToken) {
+          const token = req.header('XTM-Hub-Platform-Token');
+          if (!token) {
+            res.status(403).json({ message: 'missing token in headers' });
+            return;
```

Comment:
```suggestion
            return res.status(403).json({ message: 'missing token in headers' });
```
Should be the same ?